src/sentry/sentry_apps/api/utils/webhook_requests.py

```diff
@@ -0,0 +1,99 @@
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from typing import TypedDict
+
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.sentry_apps.models.sentry_app import SentryApp
+from sentry.sentry_apps.services.app_request import RpcSentryAppRequest, SentryAppRequestFilterArgs
+from sentry.sentry_apps.services.app_request.serial import serialize_rpc_sentry_app_request
+from sentry.sentry_apps.services.app_request.service import app_request_service
+from sentry.types.region import find_all_region_names
+from sentry.utils.sentry_apps import SentryAppWebhookRequestsBuffer
+
+
+@dataclass
+class BufferedRequest:
+    id: int
+    data: RpcSentryAppRequest
+
+    def __hash__(self):
+        return self.id
+
+
+class DatetimeOrganizationFilterArgs(TypedDict):
+    start_time: datetime
+    end_time: datetime
+    organization: OrganizationMapping | None
+
+
+def _filter_by_date(request: RpcSentryAppRequest, start: datetime, end: datetime) -> bool:
+    date_str = request.date
+    if not date_str:
+        return False
+    timestamp = datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S.%f+00:00").replace(
+        microsecond=0, tzinfo=timezone.utc
+    )
+    return start <= timestamp <= end
+
+
+def _filter_by_organization(
+    request: RpcSentryAppRequest, organization: OrganizationMapping | None
+) -> bool:
+    if not organization:
+        return True
+    return request.organization_id == organization.organization_id
+
+
+def filter_requests(
+    unfiltered_requests: list[RpcSentryAppRequest],
+    filter: DatetimeOrganizationFilterArgs,
+) -> list[BufferedRequest]:
+    requests: list[BufferedRequest] = []
+    for i, req in enumerate(unfiltered_requests):
+        start_time = filter.get("start_time")
+        end_time = filter.get("end_time")
+        if (
+            start_time
+            and end_time
+            and _filter_by_date(req, start_time, end_time)
+            and _filter_by_organization(req, organization=filter.get("organization"))
+        ):
+            requests.append(BufferedRequest(id=i, data=req))
+    return requests
+
+
+def get_buffer_requests_from_control(
+    sentry_app: SentryApp,
+    filter: SentryAppRequestFilterArgs,
+    datetime_org_filter: DatetimeOrganizationFilterArgs,
+) -> list[BufferedRequest]:
+    control_buffer = SentryAppWebhookRequestsBuffer(sentry_app)
+
+    event = filter.get("event", None) if filter else None
+    errors_only = filter.get("errors_only", False) if filter else False
```

Comment:
What are the cases filter can be None ? I think empty TypedDict is just an empty dict so we shouldn't need the ternary
```suggestion
    event = filter.get("event", None)
    errors_only = filter.get("errors_only", False)
```